server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1047 @@
+
+        if missing_params:
+            raise MissingValue(missing_params)
+
+        agent_manager = factory.consumer_agent_manager()
+        task = agent_manager.uninstall_content(consumer_id, units, options)
+        raise OperationPostponed(TaskResult.from_task_status_dict(task))
+
+
+class ConsumerHistoryView(View):
+    """
+    View for consumer history retrieval.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, consumer_id):
+        """
+        Retrieve histroy for the specified consumer
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+
+        :return: Response representing the binding
+        :rtype: django.http.HttpResponse
+        """
+        query_param = request.GET
+        filters = _ensure_input_encoding(query_param)
+        event_type = filters.get('event_type', None)
+        limit = filters.get('limit', None)
+        sort = filters.get('sort', None)
+        start_date = filters.get('start_date', None)
+        end_date = filters.get('end_date', None)
+
+        if sort is None:
+            sort = 'descending'
+
+        if limit:
+            limit = int(limit)
+
+        results = factory.consumer_history_manager().query(consumer_id=consumer_id,
+                                                           event_type=event_type,
+                                                           limit=limit,
+                                                           sort=sort,
+                                                           start_date=start_date,
+                                                           end_date=end_date)
+        if results:
+            return generate_json_response_with_pulp_encoder(results)
+        else:
+            msg = 'No history found for: %(r)s' % {'r': consumer_id}
+            return HttpResponseNotFound(msg)
+
+
+class ConsumerProfilesView(View):
+    """
+    View Consumer profiles represents the collection of
+    objects used to associate consumers and installed content
+    unit profiles.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, consumer_id):
+        """
+        Get all profiles associated with a consumer.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+
+        :return: Response representing list of profiles
+        :rtype: django.http.HttpResponse
+        """
+        # Check that the consumer exists and raise a MissingResource exception, in case it doesn't.
+        factory.consumer_manager().get_consumer(consumer_id)
+
+        manager = factory.consumer_profile_manager()
+        profiles = manager.get_profiles(consumer_id)
+        for profile in profiles:
+            add_link_profile(profile)
```

Comment:
On 03/25/2015 04:46 PM, Austin wrote:
> :punch: @rbarlow <https://github.com/rbarlow>

Ouch! That's the second punch in one day!

-- 
Randy Barlow